cmd/preflight/cmd/check_container_test.go

```diff
@@ -16,22 +22,90 @@ import (
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/internal/viper"
 
 	"github.com/go-logr/logr"
+	"github.com/google/go-containerregistry/pkg/crane"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/registry"
+	cranev1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/random"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
 )
 
 var _ = Describe("Check Container Command", func() {
+	var src string
+	var manifestListSrc string
+	var manifests map[string]string
+	var s *httptest.Server
+	var u *url.URL
+	BeforeEach(func() {
+		manifests = make(map[string]string, 2)
+		// Set up a fake registry.
+		registryLogger := log.New(io.Discard, "", log.Ldate)
+		s = httptest.NewServer(registry.New(registry.Logger(registryLogger)))
+		DeferCleanup(s.Close)
+
+		var err error
+		u, err = url.Parse(s.URL)
+		Expect(err).ToNot(HaveOccurred())
+
+		src = fmt.Sprintf("%s/test/crane", u.Host)
+		manifests["image"] = src
+
+		// Expected values.
+		img, err := random.Image(1024, 5)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = crane.Push(img, src)
+		Expect(err).ToNot(HaveOccurred())
+
+		platforms := [4]string{"x86_64", "arm64", "ppc64le", "s390x"}
+		manifestListSrc = fmt.Sprintf("%s/test/cranelist", u.Host)
+		manifests["index"] = manifestListSrc
+		lst, err := random.Index(1024, 5, int64(len(platforms)))
+		Expect(err).ToNot(HaveOccurred())
+
+		ref, err := name.ParseReference(manifestListSrc)
+		Expect(err).ToNot(HaveOccurred())
+
+		m, err := lst.IndexManifest()
+		Expect(err).ToNot(HaveOccurred())
+
+		for i, manifest := range m.Manifests {
+			switch {
+			case manifest.MediaType.IsImage():
+				m.Manifests[i].Platform = &cranev1.Platform{
+					Architecture: platforms[i],
+					OS:           "Linux",
```

Comment:
Does case matter on this? In our code I believe we have lower case ie `linux`.

Reply:
Probably not, but it should be lower. Thanks.